lms/services/organization.py

```diff
@@ -43,29 +43,38 @@ def get_by_linked_guid(self, guid) -> List[Organization]:
     def get_by_public_id(self, public_id: str) -> Optional[List]:
         """Get an organization by its public_id."""
 
-        return (
-            self._db_session.query(Organization)
-            .filter(Organization.public_id_eq(public_id, self._region))
-            .one_or_none()
-        )
+        return self._search_query(public_id=public_id).one_or_none()
 
-    def search(self, name, limit=100):
+    def search(self, public_id=None, name=None, limit=100) -> List[Organization]:
         """
-        Search organizations.
+        Search for organizations.
+
+        The results are returned as an OR of the specified filters.
 
-        :param name: Match organization by name. Case insensitive.
+        :param public_id: Match on public id
+        :param name: Match organization by name. Case-insensitive.
         :param limit: Limit the number of results
         """
-        return (
-            self._db_session.query(Organization)
-            .filter(
+        return self._search_query(public_id=public_id, name=name).limit(limit).all()
+
+    def _search_query(self, public_id=None, name=None):
+        clauses = []
+
+        if public_id:
+            clauses.append(Organization.public_id_eq(public_id, self._region))
+
+        if name:
+            clauses.append(
                 sa.func.to_tsvector("english", Organization.name).op("@@")(
                     sa.func.websearch_to_tsquery(name, postgresql_regconfig="english")
                 )
             )
-            .limit(limit)
-            .all()
-        )
+
+        query = self._db_session.query(Organization)
+        if clauses:
+            query = query.filter(sa.or_(*clauses))
```

Comment:
It would be very easy to parameterise the `sa.or_` here to an `and` if there was ever a need.